plugins/inputs/snmp/snmp.go

```diff
@@ -120,13 +98,17 @@ type Snmp struct {
 
 	connectionCache []snmpConnection
 	initialized     bool
+
+	Log telegraf.Logger `toml:"-"`
 }
 
 func (s *Snmp) init() error {
```

Comment:
Not directly related to the PR but we should convert this to `Init()` and remove the `init` logic...

Reply:
I like this suggestion but think it would be best in another pr. What do you think of that?

Reply:
I agree with you. Let's shift this to another PR.

Reply:
While restructuring getMibsPath this `init` got refactored into `Init()` so no need for another pr now :)